libraries/aws_iam_access_key.rb

```diff
@@ -13,94 +13,94 @@ class AwsIamAccessKey < Inspec.resource(1)
     end
   "
 
-  def initialize(opts, decorator = IamClientDecorator.new)
-    @access_key = opts[:access_key]
-    @username = opts[:username]
-    @id = @access_key ? @access_key.access_key_id : opts[:id]
-
-    @decorator = decorator
-  end
+  include AwsSingularResourceMixin
+  attr_reader :username, :access_key_id, :status, :create_date
```

Comment:
Nit: I am fan of ordering these by alpha as it makes it easier to find things if they grow.